### ISSUE TITLE
Dockerfile compile error: Cannot select: 0x2045850: v8i64 = X86ISD::PCMPEQ 0x20455e0, 0x701e38

### ISSUE THREAD

**Issue by blurayne** [Genivia/ugrep#290]
Using Docker container for build which is Ubuntu 22.04.3 LTS. Same in podman as with docker (rootless and root):

```bash
$ lscpu
Model name:            11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
```

```
clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
mv -f .deps/libreflex_a-block_scripts.Tpo .deps/libreflex_a-block_scripts.Po
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/simd_avx512bw-d99dd5.cpp
clang: note: diagnostic msg: /tmp/simd_avx512bw-d99dd5.sh
clang: note: diagnostic msg: 

********************
make[2]: *** [Makefile:561: libreflex_a-simd_avx512bw.o] Error 70
make[2]: *** Waiting for unfinished jobs....
mv -f .deps/libreflex_a-convert.Tpo .deps/libreflex_a-convert.Po
mv -f .deps/libreflex_a-matcher.Tpo .deps/libreflex_a-matcher.Po
mv -f .deps/libreflex_a-matcher_avx2.Tpo .deps/libreflex_a-matcher_avx2.Po
mv -f .deps/libreflex_a-matcher_avx512bw.Tpo .deps/libreflex_a-matcher_avx512bw.Po
mv -f .deps/libreflex_a-pattern.Tpo .deps/libreflex_a-pattern.Po
make[2]: Leaving directory '/work/lib'
make[1]: *** [Makefile:464: all-recursive] Error 1
make[1]: Leaving directory '/work'
make: *** [Makefile:364: all] Error 2
Failed to build ugrep: please run the following two commands:
$ autoreconf -fi
$ ./build.sh

If that does not work, please open an issue at:
https://github.com/Genivia/ugrep/issues
```

`/tmp/simd_avx512bw-d99dd5.sh`

```
# Crash reproducer for Ubuntu clang version 14.0.0-1ubuntu1.1
# Driver args: "--driver-mode=g++" "-std=gnu++11" "-D" "HAVE_CONFIG_H" "-I" "." "-I" ".." "-I" "../include" "-msse2" "-D" "HAVE_AVX512BW" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-Wall" "-Wextra" "-Wunused" "-O2" "-mavx512bw" "-MT" "libreflex_a-simd_avx512bw.o" "-MD" "-MP" "-MF" ".deps/libreflex_a-simd_avx512bw.Tpo" "-c" "-o" "libreflex_a-simd_avx512bw.o" "simd_avx512bw.cpp"
# Original command:  "/usr/lib/llvm-14/bin/clang" "-cc1" "-triple" "x86_64-pc-linux-gnu" "-emit-obj" "--mrelax-relocations" "-disable-free" "-clear-ast-before-backend" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "simd_avx512bw.cpp" "-mrelocation-model" "pic" "-pic-level" "2" "-pic-is-pie" "-mframe-pointer=none" "-fmath-errno" "-ffp-contract=on" "-fno-rounding-math" "-mconstructor-aliases" "-funwind-tables=2" "-target-cpu" "x86-64" "-target-feature" "+sse2" "-target-feature" "+avx512bw" "-tune-cpu" "generic" "-mllvm" "-treat-scalable-fixed-error-as-warning" "-debugger-tuning=gdb" "-fcoverage-compilation-dir=/work/lib" "-resource-dir" "/usr/lib/llvm-14/lib/clang/14.0.0" "-dependency-file" ".deps/libreflex_a-simd_avx512bw.Tpo" "-MT" "libreflex_a-simd_avx512bw.o" "-sys-header-deps" "-MP" "-D" "HAVE_CONFIG_H" "-I" "." "-I" ".." "-I" "../include" "-D" "HAVE_AVX512BW" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-I" "/usr/include" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/backward" "-internal-isystem" "/usr/lib/llvm-14/lib/clang/14.0.0/include" "-internal-isystem" "/usr/local/include" "-internal-isystem" "/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include" "-internal-externc-isystem" "/usr/include/x86_64-linux-gnu" "-internal-externc-isystem" "/include" "-internal-externc-isystem" "/usr/include" "-O2" "-Wall" "-Wextra" "-Wunused" "-std=gnu++11" "-fdeprecated-macro" "-fdebug-compilation-dir=/work/lib" "-ferror-limit" "19" "-fgnuc-version=4.2.1" "-fcxx-exceptions" "-fexceptions" "-fcolor-diagnostics" "-vectorize-loops" "-vectorize-slp" "-faddrsig" "-D__GCC_HAVE_DWARF2_CFI_ASM=1" "-o" "libreflex_a-simd_avx512bw.o" "-x" "c++" "simd_avx512bw.cpp"
 "/usr/lib/llvm-14/bin/clang" "-cc1" "-triple" "x86_64-pc-linux-gnu" "-emit-obj" "--mrelax-relocations" "-disable-free" "-clear-ast-before-backend" "-disable-llvm-verifier" "-discard-value-names" "-main-file-name" "simd_avx512bw.cpp" "-mrelocation-model" "pic" "-pic-level" "2" "-pic-is-pie" "-mframe-pointer=none" "-fmath-errno" "-ffp-contract=on" "-fno-rounding-math" "-mconstructor-aliases" "-funwind-tables=2" "-target-cpu" "x86-64" "-target-feature" "+sse2" "-target-feature" "+avx512bw" "-tune-cpu" "generic" "-mllvm" "-treat-scalable-fixed-error-as-warning" "-debugger-tuning=gdb" "-fcoverage-compilation-dir=/work/lib" "-sys-header-deps" "-D" "HAVE_CONFIG_H" "-D" "HAVE_AVX512BW" "-O2" "-Wall" "-Wextra" "-Wunused" "-std=gnu++11" "-fdeprecated-macro" "-fdebug-compilation-dir=/work/lib" "-ferror-limit" "19" "-fgnuc-version=4.2.1" "-fcxx-exceptions" "-fexceptions" "-fcolor-diagnostics" "-vectorize-loops" "-vectorize-slp" "-faddrsig" "-D__GCC_HAVE_DWARF2_CFI_ASM=1" "-x" "c++" "simd_avx512bw-d99dd5.cpp"
```

**Now digging for the real error message:**

```bash
root@54a0f4b02cb7:/tmp# sh simd_avx512bw-d99dd5.sh 
fatal error: error in backend: Cannot select: 0x2045850: v8i64 = X86ISD::PCMPEQ 0x20455e0, 0x701e38
  0x20455e0: v8i64,ch = load<(load (s64) from %ir.25, align 1, !tbaa !9), zext from v8i8> 0xa46078, 0x2045988, undef:i64
    0x2045988: i64 = add 0x2044f60, Constant:i64<8>
      0x2044f60: i64,ch = CopyFromReg 0xa46078, Register:i64 %0
        0x701ea0: i64 = Register %0
      0x701d00: i64 = Constant<8>
    0x2045308: i64 = undef
  0x701e38: v8i64,ch = X86ISD::VBROADCAST_LOAD<(load (s64) from constant-pool)> 0xa46078, 0x20457e8
    0x20457e8: i64 = X86ISD::WrapperRIP TargetConstantPool:i64<i64 10> 0
      0x2045bf8: i64 = TargetConstantPool<i64 10> 0
In function: _ZN6reflex21simd_nlcount_avx512bwERPKcS1_
` ` `


[simd_avx512bw-d99dd5.cpp.gz](https://github.com/Genivia/ugrep/files/12474030/simd_avx512bw-d99dd5.cpp.gz)


Basically I should have AVX512… 

Compiling on my host system which curently is also Ubutntu 22 worked fine. 

Maybe a package missing (maybe not) or something messed up with the CPU detection inside the container?

**Comment by blurayne** [comment]
Something LLVM related? Or Docker/Podman? Or Kernel… hmmm… using xanmod 6.4 kernel currently.

```
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: c++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I.. -I../include -msse2 -DHAVE_AVX512BW -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -Wall -Wextra -Wunused -O2 -mavx512bw -MT libreflex_a-simd_avx512bw.o -MD -MP -MF .deps/libreflex_a-simd_avx512bw.Tpo -c -o libreflex_a-simd_avx512bw.o simd_avx512bw.cpp
1.	<eof> parser at end of file
2.	Code generation
3.	Running pass 'Function Pass Manager' on module 'simd_avx512bw.cpp'.
4.	Running pass 'X86 DAG->DAG Instruction Selection' on function '@_ZN6reflex21simd_nlcount_avx512bwERPKcS1_'
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys15PrintStackTraceERNS_11raw_ostreamEi+0x31)[0x7f6986900d01]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys17RunSignalHandlersEv+0xee)[0x7f69868fea3e]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys15CleanupOnSignalEm+0xfb)[0x7f69869000ab]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xd6bbea)[0x7f698682cbea]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xd6bb8b)[0x7f698682cb8b]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys7Process4ExitEib+0x27)[0x7f69868fb007]
c++[0x4136d2]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm18report_fatal_errorERKNS_5TwineEb+0x113)[0x7f698683b673]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0x1660e31)[0x7f6987121e31]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm16SelectionDAGISel16SelectCodeCommonEPNS_6SDNodeEPKhj+0x3908)[0x7f69871212d8]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0x3600c1a)[0x7f69890c1c1a]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm16SelectionDAGISel22DoInstructionSelectionEv+0x19f)[0x7f6987119ecf]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm16SelectionDAGISel17CodeGenAndEmitDAGEv+0x5c4)[0x7f6987119594]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm16SelectionDAGISel20SelectAllBasicBlocksERKNS_8FunctionE+0x1778)[0x7f69871189c8]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm16SelectionDAGISel20runOnMachineFunctionERNS_15MachineFunctionE+0x8c7)[0x7f6987116557]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0x35f7a0d)[0x7f69890b8a0d]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm19MachineFunctionPass13runOnFunctionERNS_8FunctionE+0x12e)[0x7f6986c8728e]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE+0x3a0)[0x7f6986a3b390]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE+0x33)[0x7f6986a42983]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE+0x946)[0x7f6986a3bf36]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang17EmitBackendOutputERNS_17DiagnosticsEngineERKNS_19HeaderSearchOptionsERKNS_14CodeGenOptionsERKNS_13TargetOptionsERKNS_11LangOptionsEN4llvm9StringRefEPNSE_6ModuleENS_13BackendActionESt10unique_ptrINSE_17raw_pwrite_streamESt14default_deleteISK_EE+0xbc1)[0x7f698dbea201]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(+0x1b7c855)[0x7f698df0f855]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang8ParseASTERNS_4SemaEbb+0x244)[0x7f698cd97824]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang13CodeGenAction13ExecuteActionEv+0xb1)[0x7f698df0bb71]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang14FrontendAction7ExecuteEv+0x67)[0x7f698e8a7b57]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang16CompilerInstance13ExecuteActionERNS_14FrontendActionE+0x336)[0x7f698e7ff3a6]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang25ExecuteCompilerInvocationEPNS_16CompilerInstanceE+0x29b)[0x7f698e92145b]
c++(_Z8cc1_mainN4llvm8ArrayRefIPKcEES2_Pv+0x99b)[0x41328b]
c++[0x4114bc]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(+0x20eaed2)[0x7f698e47ded2]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm20CrashRecoveryContext9RunSafelyENS_12function_refIFvvEEE+0xdd)[0x7f698682cb6d]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZNK5clang6driver10CC1Command7ExecuteEN4llvm8ArrayRefINS2_8OptionalINS2_9StringRefEEEEEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPb+0x140)[0x7f698e47d9c0]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZNK5clang6driver11Compilation14ExecuteCommandERKNS0_7CommandERPS3_+0x3f3)[0x7f698e448183]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZNK5clang6driver11Compilation11ExecuteJobsERKNS0_7JobListERN4llvm15SmallVectorImplISt4pairIiPKNS0_7CommandEEEE+0x8a)[0x7f698e44840a]
/lib/x86_64-linux-gnu/libclang-cpp.so.14(_ZN5clang6driver6Driver18ExecuteCompilationERNS0_11CompilationERN4llvm15SmallVectorImplISt4pairIiPKNS0_7CommandEEEE+0x1a7)[0x7f698e462507]
c++(main+0x2816)[0x410f26]
/lib/x86_64-linux-gnu/libc.so.6(+0x29d90)[0x7f698558dd90]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80)[0x7f698558de40]
c++(_start+0x25)[0x40e3b5]
clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
mv -f .deps/libreflex_a-simd_avx2.Tpo .deps/libreflex_a-simd_avx2.Po
clang: note: diagnostic msg
```

**Comment by genivia-inc** [comment]
Thank you for your feedback and for sharing the details.

You don't need AVX512 to compile and run ugrep. It detects SSE2/AVX2/AVX512 at compile time and at runtime for binary portability on x64 systems.

The source `simd_avx512bw.cpp` is compiled with AVX512BW intrinsics only if `HAVE_AVX512BW` is defined. This suggests an arch/mach detection problem when the ugrep build is configured.

The `./configure` output produced by `./build.sh` should report SSE2, AVX2, AVX512 support when available.

Does AVX512 show up as supported in the "checking for..." output?

**Comment by genivia-inc** [comment]
Please provide the output produced by `./build.sh` as suggested above, so we know where and when this happens. If `./build.sh` is executed correctly then there should not be a problem like this. But if `./build.sh` reports AVX512 when the machine doesn't support it, then it might be a clang/gcc/llvm problem with the `-march=native` option.
